exist-core/src/main/java/org/exist/validation/resolver/eXistXMLCatalogResolver.java

```diff
@@ -49,6 +49,10 @@
  */
 public class eXistXMLCatalogResolver extends XMLCatalogResolver {
 
+    //TODO(AR) refactor to org.apache.xml.resolver.tools.CatalogResolver which is reused within here for Xerces
```

Comment:
I'd suggest to have a look at Norman's recent work on Catalog resolvers and extend/contribute to that initiative.